simplicity-sys/src/c_jets/frame_ffi.rs

```diff
@@ -0,0 +1,27 @@
+//! Frame related FFI bindings and data structures
+
+use std::os::raw::c_uchar;
+
+use libc::size_t;
+
+/// FrameItem for C FFI.
+/// This type is only used to pass data to the C FFI.
+#[derive(Debug)]
+#[repr(C)]
+pub struct CFrameItem {
+    pub edge: *const c_uchar,
+    pub len: size_t,
+}
+
+extern "C" {
+    pub static c_sizeof_frameItem: size_t;
+    pub static c_sizeof_UWORD: size_t;
+
+    pub(crate) fn c_initWriteFrame(frame: &mut CFrameItem, n: size_t, from: *const c_uchar);
+    pub(crate) fn c_initReadFrame(frame: &mut CFrameItem, n: size_t, from: *const c_uchar);
```

Comment:
In 59c2942084530ec7c2d06b9147dccd3156aa51c3:

I'm pretty suspicious about using `c_uchar` instead of `u16` here. The alignment requirements are different.

Reply:
`c_uchar` instead of `u8` you mean?

Reply:
The caller has to take care about calling with the correct alignment of `c_uchar` which we do in `c_frame.rs`. Refer to the functions `round_u_word` and `ffi_bytes_size`

Reply:
No, `c_uchar` and `u8` are the same, that would be fine. The issue is that the C code is expecting a pointer to u16 while we're giving it a pointer to u8.

It looks like `round_u_word` is only ever called from `ffi_bytes_size`, which is never called.

Reply:
See example usage: https://github.com/sanket1729/rust-simplicity/blob/911b6072f90341936479948c08e22c7cd20aa70f/src/jet/mod.rs#L126-L132

Is alignment a concern here? We are passing a pointer to `u8 * sizeof(UWORD)` instead of `UWORD`?

Reply:
Perhaps, all this should be a reference to `usize`? I think on all systems, but 8 bit systems, `uint_fast_16` should map to usize?

Reply:
>These functions are not called in this PR. We are not using jets here.

If you are required to call a separate alignment function to avoid UB then these functions can't be marked `safe`.

>Perhaps, all this should be a reference to usize? I think on all systems, but 8 bit systems, uint_fast_16 should map to usize?

I think this would be okay, though we should have tests that assert this.